frontend/src/features/projects/common/projectConverter.ts

```diff
@@ -90,66 +90,75 @@ export const getFlatProjectNotes = (project: IApiProject) => {
   return notes;
 };
 
+export const toFlatProperty = (apiProperty: IApiProperty, pp?: IProjectProperty): IProperty => {
```

Comment:
Split this out into a separate function to simplify the logic and allow direct testing.